pkg/gui/controllers/commits_files_controller.go

```diff
@@ -153,13 +153,24 @@ func (self *CommitFilesController) checkout(node *filetree.CommitFileNode) error
 }
 
 func (self *CommitFilesController) discard(node *filetree.CommitFileNode) error {
+	if node.File == nil {
```

Comment:
I explained this in the commit message. Discarding whole directories does work as long as the changes are only modifications to existing files. It doesn't work correctly when files were added or deleted. It would probably be possible to fix this, by using something similar to a custom patch under the hood, but I'm not sure it's worth it.

Reply:
Ah sorry: I'm used to getting all the context from the PR description at my day job (my work isn't big on writing detailed commit messages). I'll be sure to read the commits next time!

I'm happy for us to raise an error for now: better to be explicit

Reply:
I added the error in b4ddb12b5359.